### DISCUSSION TITLE
Showing installed version as incorrect one

### DISCUSSION THREAD

**Discussion by gayathrisj** [aquasecurity/trivy#5024] [Q&A] [answered]
### Description

I have been using the version: trivy_0.42.0_Linux-64bit.tar.gz
we had issues in a specific version of the package, then we upgraded to the fixed version, still, it is fetching the old version as the installed version and showing the error

### Desired Behavior

It should show the installed version as the version that we have upgraded eg. ansi-regex as 3.0.1, 5.0.1, 6.0.1	

### Actual Behavior

we are getting the incorrectly installed version as follows
![image](https://github.com/aquasecurity/trivy/assets/21053202/f310a60a-9294-40a3-a5ce-4031ae9a364c)

### Reproduction Steps

```bash
1. install the package ansi-regex as 3.0.1 in dockerfile
2. scan dockerfile for vulnerabilities
```

### Target

None

### Scanner

Vulnerability

### Output Format

Template

### Mode

Client/Server

### Debug Output

```bash
we used ./trivy image --no-progress --ignore-unfixed --exit-code 1 --severity CRITICAL,HIGH engine
```

### Operating System

Ubuntu

### Version

```bash
v0.44.0
```

### Checklist

- [X] Run `trivy image --reset`
- [X] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)

**Comment by DmitriyLewen** [comment]
Hello @gayathrisj 
Thanks for your report!

Can you scan your image with `-f json --list-all-pkgs` flags and share `ansi-regex` packages(`Results.Packages[]`) with wrong version?

Regards, Dmitriy

**Comment by gayathrisj** [comment]
Hi @DmitriyLewen 
Thanks for your response.
Please find the key you have asked
        {
          "created": "2023-08-22T07:59:03Z",
          "created_by": "RUN /bin/sh -c yarn add ansi-regex@3.0.1 http-cache-semantics@4.1.1 minimatch@3.0.5 npm@8.11.0 # buildkit",
          "comment": "buildkit.dockerfile.v0"
        }

**Reply by DmitriyLewen** [reply]
What base image do you use?
I tried with `node:latest` and didn't find these CVEs:
```
➜ cat Dockerfile 
FROM node

RUN mkdir app && cd app
RUN yarn add ansi-regex@3.0.1 http-cache-semantics@4.1.1 minimatch@3.0.5 npm@8.11.0%                                                                                                      

➜ trivy -d -q image 5024 | grep ansi-regex
```

Perhaps another project uses old `ansi-regex`?
You can see path to `yarn.lock` files with `-f json --list-all-pkgs` flag.

**Comment by gayathrisj** [comment]
we use node:16.14.0

**Reply by DmitriyLewen** [reply, marked as answer]
`node:16.14.0` contains `package.json` files with non-`ansi-regex@3.0.1`:
```
➜ trivy -d -q image -f json --list-all-pkgs node:16.14.0 | grep '"ID": "ansi-regex@' -A 10 | grep '"ID"\|"FilePath"'
          "ID": "ansi-regex@2.1.1",
          "FilePath": "usr/local/lib/node_modules/npm/node_modules/ansi-regex/package.json"
          "ID": "ansi-regex@5.0.0",
          "FilePath": "usr/local/lib/node_modules/npm/node_modules/cli-table3/node_modules/ansi-regex/package.json"
          "ID": "ansi-regex@3.0.0",
          "FilePath": "usr/local/lib/node_modules/npm/node_modules/string-width/node_modules/ansi-regex/package.json"
          "ID": "ansi-regex@5.0.1",
          "FilePath": "usr/local/lib/node_modules/npm/node_modules/cli-columns/node_modules/ansi-regex/package.json"
          "ID": "ansi-regex@5.0.1",
          "FilePath": "usr/local/lib/node_modules/npm/node_modules/gauge/node_modules/ansi-regex/package.json"
```
e.g.:
```
➜ docker run -it --rm --entrypoint /bin/bash node:16.14.0
...

root@d35a1df197a1:/# cat usr/local/lib/node_modules/npm/node_modules/ansi-regex/package.json | grep '"name": "ansi-regex",' -A 1
  "name": "ansi-regex",
  "version": "2.1.1",
```

**Reply by gayathrisj** [reply]
thanks

**Reply by DmitriyLewen** [reply]
you can remove these packages or use [skip-dirs](https://aquasecurity.github.io/trivy/v0.44/docs/configuration/skipping/#skip-directories).
